quickjs.c

```diff
@@ -40370,7 +40370,40 @@ static JSValue js_iterator_proto_some(JSContext *ctx, JSValue this_val,
 static JSValue js_iterator_proto_take(JSContext *ctx, JSValue this_val,
                                       int argc, JSValue *argv)
 {
-    return JS_ThrowInternalError(ctx, "TODO implement Iterator.prototype.take");
+    JSValue v;
+    double dlimit;
+    int64_t limit;
+    if (!JS_IsObject(this_val))
+        return JS_ThrowTypeError(ctx, "Iterator.prototype.take called on non-object");
+    v = JS_ToNumber(ctx, argv[0]);
+    if (JS_IsException(v))
+        return JS_EXCEPTION;
+    // Check for Infinity.
+    if (JS_ToFloat64(ctx, &dlimit, v)) {
+        JS_FreeValue(ctx, v);
+        return JS_EXCEPTION;
+    }
+    if (isnan(dlimit)) {
+        JS_FreeValue(ctx, v);
+        goto fail;
+    }
+    if (!isfinite(dlimit)) {
+        JS_FreeValue(ctx, v);
+        if (dlimit < 0)
+            goto fail;
+        else
+            limit = MAX_SAFE_INTEGER;
+    } else {
+        v = JS_ToIntegerFree(ctx, v);
+        if (JS_IsException(v))
+            return JS_EXCEPTION;
+        if (JS_ToInt64Free(ctx, &limit, v))
+            return JS_EXCEPTION;
+    }
+    if (limit < 0)
+    fail:
+        return JS_ThrowRangeError(ctx, "must be positive");
```

Comment:
```suggestion
    if (limit < 0) {
    fail:
        return JS_ThrowRangeError(ctx, "must be positive");
    }
```
(dominant style for this kind of thing)